### PR TITLE
feat: editable select + some fixes

### PR DESCRIPTION
**Pull request by mjashanks**
### Editable Select:

https://www.loom.com/share/4fe76809071d4f03925ef0e41ddf26b5

Ability to free type into a select... usage

```
<Select editable >
....
</Select>
```

### Dropdown Fix

When a dropdownMenu was used inside a modal, it was positioned way on the other side of the screen

### Modal Hieght/Width

Modal had a min height and width - this was causing content to something not fit, or scroll unecessarily

### CODE REVIEW

File: src/Form/Select.svelte [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
   export let outline = false;
   export let disabled = false;
-  import Icon from "../Icons/Icon.svelte";
+  export let editable = false;
+  
+  const dispatch = createEventDispatcher();
+
+  const updateValue = type => e => {
+    value = e.target.value;
+    dispatch("blur", e)
```
Comment (Conor-Mack) [inline src/Form/Select.svelte:17]: This is nice, although should it be `dispatch(type, e)`

File: src/Form/Select.svelte [hunk 2]
```diff
@@ -54,24 +84,49 @@
     bottom: 0 !important;
```
Comment (Conor-Mack) [inline src/Form/Select.svelte:84]: How come so many `!important` rules?

File: src/Form/Select.svelte [hunk 3]
```diff
@@ -54,24 +84,49 @@
...
     bind:value>
     <slot />
   </select>
-  <div class="pointer">
+  {#if editable}
+    <input  
+      class:thin 
+      class:secondary 
+      class:disabled
+      on:change={updateValue("change")}
```
Comment (shogunpurple) [inline src/Form/Select.svelte:122]: I think we might be able to get rid of the event dispatcher and curried function and just bind the input the same way we are with the `select` here.

```
on:input
on:blur
bind:value
```

**Review by shogunpurple — CHANGES_REQUESTED** [review]
Just the wee issue conor mentioned. We can potentially remove the event dispatcher stuff as well, provided there's no weird clash when you bind to 2 input/select elements.

File: src/Form/Select.svelte (hunk 1 above)
Comment (mjashanks) [reply src/Form/Select.svelte:17]: yes, good one cheers

File: src/Form/Select.svelte (hunk 2 above)
Comment (mjashanks) [reply src/Form/Select.svelte:84]: They came with the select - I just left them alone! Don't anger the CSS gods!

File: src/Form/Select.svelte (hunk 3 above)
Comment (mjashanks) [reply src/Form/Select.svelte:122]: Yes - it was like this... but was not working... the blur event was not firing (?)

Comment (shogunpurple) [reply src/Form/Select.svelte:122]: Ah - I think this might be because the `blur` event doesn't bubble.

Comment (mjashanks) [reply src/Form/Select.svelte]: updated so only on:blur does a `dispatch`

**Review by shogunpurple — APPROVED** [review]

**Pull request merged**